test/k8s-local-cluster-test/provision-eks-sqs-infra

```diff
@@ -0,0 +1,386 @@
+#!/bin/bash
+
+set -euo pipefail
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+export AWS_PAGER=""
+export AWS_REGION=""
+ENDPOINT_URL="" #"--endpoint-url=http://localhost:4566" <- This is for localstack
+EXTRA_AWS_PARAMS=""
+DELETE=0
+FLEET=0
+ACCOUNT_ID=""
+CLUSTER_NAME=""
+SQS_QUEUE_NAME="MyK8sTermQueue"
+EVENTBRIDGE_ASG_RULE_NAME="MyK8sASGTermRule"
+EVENTBRIDGE_SPOT_RULE_NAME="MyK8sSpotTermRule"
+EVENTBRIDGE_EC2_STATE_CHANGE_RULE_NAME="MyK8sEC2StateChangeRule"
+LIFECYCLE_HOOK_NAME="my-k8s-term-hook"
+IAM_POLICY="term-event-policy"
+ROLE_NAME="term-event-role"
+POLICY_NAME="cth"
+RAND_ID=$(uuidgen | cut -d'-' -f1 | tr '[:upper:]' '[:lower:]')
+VERSION="v1.0.0-${RAND_ID}"
+SERVICE_ACCOUNT_NAME="aws-cluster-termination-handler-aws-node-termination-handler"
+
+USAGE=$(cat << EOM
+  Usage: `basename $0` [-n <name>] [-s] [-w] [-h]
+  Installs NTH from a local repo on a k8s cluster .
+  The script also provisions SQS, EventBridge, and EC2 ASG Lifecycle Hook resources.
+
+  Example: cth-setup -r my-test-cluster
+
+          Required:
+            -n          Name of the cluster (default is a random ID)
+            -r          Region (default us-east-2)
+
+          Optional:
+            -a          Account ID
+            -s          SQS Queue Name
+            -f          Add a Spot Fleet Node Group
+            -d          Delete infrastructure
+            -v          Verbose output
+            -h          Show this usage help
+EOM
+)
+
+usage() {
+  echo "$USAGE" 1>&2
+  exit "$1"
+}
+
+# Process our input arguments
+while getopts "hn:r:a:s:vdf" opt; do
+  case ${opt} in
+    a ) # Account ID
+        ACCOUNT_ID="${OPTARG}"
+      ;;
+    n ) # Name
+        CLUSTER_NAME="${OPTARG}"
+      ;;
+    s ) # SQS Queue Name?
+        SQS_QUEUE_NAME="${OPTARG}"
+      ;;
+    r ) # AWS Region
+        AWS_REGION="${OPTARG}"
+      ;;
+    d ) # Delete infa
```

Comment:
nit: are you choosing delete over preserve because default behavior is to keep the infra setup?

Reply:
Yep, because this isn't an automated test workflow, it's just a helper to setup some infra for manual testing, or potentially an automated test. But the automated test would be responsible for tearing down the infrastructure by calling delete.